server/routes/user.js

```diff
@@ -32,111 +28,20 @@ router.post('/api/user', (req, res) => {
-
-  return axios.all([
-    getFrontEndCert(username),
-    getBackEndCert(username),
-    getDataVisCert(username)
-  ]).then(axios.spread((frontCert, backCert, dataCert) => {
-    let totalRedirects =
-    frontCert.request._redirectCount +
-    backCert.request._redirectCount +
-    dataCert.request._redirectCount;
-    if (honoraryMember || totalRedirects < 3) {
-      return {
-        Front_End: frontCert.request._redirectCount === 0 ? true : false,
-        Back_End: backCert.request._redirectCount === 0 ? true : false,
-        Data_Visualization: dataCert.request._redirectCount === 0 ? true : false,
-      }
-    } else {
-      if (isAllowedForDev) {
-        return {
-          Front_End: false,
-          Back_End: false,
-          Data_Visualization: false,
-        }
-      } else {
-        return false;
-      }
-    }
-  }))
-}
-
-const handleProcessedUser = (certs, mongoId, req, res, username) => {
-  if (!certs) {
-    // user not verified, res with error
-    User.findById(mongoId, (err, user) => {
-      if (err) throw err;
-      user.verifiedUser = false;
-      user.save();
-      res.status(401).json({ error: 'User cannot be verified' });
-    });
-  } else {
-    // verified user, proceed
-    User.findById(mongoId, (err, user) => {
-      if (err) throw err;
-      /* we need to overwrite their session username too
-      (only matters for whitelisted users) */
-      req.user.username = username;
-      user.username = username;
-      user.fccCerts = certs;
-      user.verifiedUser = true;
-      user.save();
-      req.user.verifiedUser = true;
-      req.user.fccCerts = certs;
-      res.json({ user });
-    });
-  }
-}
-
-router.post('/api/update-user', (req, res) => {
-  const { user } = req.body;
-
-  User.findById(user._id, (err, updatedUser) => {
-    if (!err) {
-      updatedUser.personal = user.personal;
-      updatedUser.mentorship = user.mentorship;
-      updatedUser.career = user.career;
-      updatedUser.skillsAndInterests = user.skillsAndInterests;
-      updatedUser.projects = user.projects;
-      updatedUser.social = user.social;
-      updatedUser.save();
-      res.json({ updatedUser })
-    } else {
-      res.status(401).json({ error: 'User could not be saved' });
-    }
-  });
+  (async function processUserVerification() {
+    // if user is whitelisted, use their alternate username
+    var username = await checkWhiteList(req.body.username);
+    var isWhitelistedUser = !(username === req.body.username);
+    // if user is honorary member, they will be let in w/o certs
+    var isHonoraryMember = await checkHonoraryMemberList(username);
+    // process FCC verification...
+    isCertified(username, isHonoraryMember, isWhitelistedUser)
+    .then(certs => {
+      // update user with certs and correct status in DB
+      handleProcessedUser(certs, mongoId, req, res, username);
+    }).catch(err => console.log(err));
+  })().catch(err => console.log(err));
```

Comment:
I think making this an IIFE makes it a little harder to see what's going on. Is there a need for the function `processUserVerification` at all? Can you just `await` `checkWhiteList` and `checkHonoraryMemberList` and make the overall Express callback `async`?